connectors/webhook/src/main/java/io/camunda/connector/inbound/HttpWebhookExecutable.java

```diff
@@ -64,55 +56,52 @@ public WebhookProcessingResult triggerWebhook(WebhookProcessingPayload payload)
     }
 
     response.setBody(
-        transformRawBodyToMap(payload.rawBody(), extractContentType(payload.headers())));
+        HttpWebhookUtil.transformRawBodyToMap(
+            payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers())));
     response.setHeaders(payload.headers());
     response.setParams(payload.params());
 
     return response;
   }
 
-  private String extractContentType(Map<String, String> headers) {
-    var caseInsensitiveMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
-    caseInsensitiveMap.putAll(headers);
-    return caseInsensitiveMap.get(HttpHeaders.CONTENT_TYPE).toString();
-  }
-
-  private Map transformRawBodyToMap(byte[] rawBody, String contentTypeHeader) throws IOException {
-    if (rawBody == null) {
-      return Collections.emptyMap();
-    }
-
-    if (MediaType.FORM_DATA.toString().equalsIgnoreCase(contentTypeHeader)) {
-      String bodyAsString = new String(rawBody, StandardCharsets.UTF_8);
-      return Arrays.stream(bodyAsString.split("&"))
-          .filter(Objects::nonNull)
-          .map(param -> param.split("="))
-          .filter(param -> param.length > 1)
-          .collect(Collectors.toMap(param -> param[0], param -> param[1]));
-    } else {
-      // Do our best to parse to JSON (throws exception otherwise)
-      return objectMapper.readValue(rawBody, Map.class);
-    }
-  }
-
   private boolean webhookSignatureIsValid(
       WebhookConnectorProperties context, WebhookProcessingPayload payload)
       throws NoSuchAlgorithmException, InvalidKeyException, IOException {
-    final String shouldValidateHmac =
-        Optional.ofNullable(context.getShouldValidateHmac()).orElse(disabled.name());
-    if (enabled.name().equals(shouldValidateHmac)) {
-      final HMACSignatureValidator hmacSignatureValidator =
-          new HMACSignatureValidator(
-              payload.rawBody(),
-              payload.headers(),
-              context.getHmacHeader(),
-              context.getHmacSecret(),
-              HMACAlgoCustomerChoice.valueOf(context.getHmacAlgorithm()));
-      return hmacSignatureValidator.isRequestValid();
+    if (shouldValidateHmac(context)) {
+      byte[] signatureData;
+      if (payload.method().equalsIgnoreCase("GET")) {
+        signatureData = HttpWebhookUtil.extractSignatureDataFromParams(payload);
+      } else {
+        if (context.getHmacHeader().equalsIgnoreCase("x-twilio-signature")) {
+          signatureData = HttpWebhookUtil.extractTwilioSignatureData(payload);
+        } else {
+          signatureData = payload.rawBody();
+        }
```

Comment:
We could have a feel expression that defines how we calculate the `hmac` based on a list like 
```
=["url", "body"]
```

This way we would get rid of the twilio specific part: 
```
context.getHmacHeader().equalsIgnoreCase("x-twilio-signature")
```

For other connectors we would just define it as `=["body"]` and in the case of `GET`: `=["url", "params"]`

Reply:
Really cool, thanks @Oleksiivanov